### PR TITLE
evmrs: add BUILD.md

### PR DESCRIPTION
**Pull request by LorenzSchueler**

### CODE REVIEW

**Comment by LuisPH3** [comment]
@simonlechner I add you here to sync style and content of the rust BUILD.md with the main BUILD.md file.

File: rust/BUILD.md [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+## Testing
+
+- Rust tests
+    ```sh
+    cargo test
+    ```
+- Go tests
+    ```sh
+    cargo build --release # OR run `make` or `make tosca-rust` in Tosca directory
+    go test ../go/...
```
Comment (simonlechner) [inline rust/BUILD.md:51]: This command will run lots of tests which do not concern evmrs, this should be narrowed down.

File: rust/BUILD.md [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    cargo clean # make sure Go does not pick up the release build
+    cargo build --profile profiling
+    perf record --call-graph dwarf -F 25000 \
+        go test ../go/integration_test/interpreter \
+        --run none --bench '^Benchmark[a-zA-Z]+/./evmrs$' \
+        --timeout 1h --count 1
+    perf script | inferno-collapse-perf | inferno-flamegraph > flamegraph.svg
+    ```
+
+Also see [flamegraph.sh](./scripts/flamegraph.sh) which when provided with a list of features sets, runs the rust benchmarks and generates a flamegraph for each feature set.
```
Comment (simonlechner) [inline rust/BUILD.md:115]: This link is broken.

File: rust/BUILD.md [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+
+## Coverage
+
+Code coverage can be generated for both the Rust test and CT.
+- Rust test coverage
+    ```sh
+    cargo install cargo-llvm-cov
+    cargo llvm-cov --open
+    ```
+- CT coverage: see [coverage.sh](./scripts/coverage.sh).
```
Comment (simonlechner) [inline rust/BUILD.md:85]: This link is broken.

**Review by simonlechner — COMMENTED** [review]
Please merge PRs with scripts before or add them to this PR do not have broken links.

I would move Testing and Coverage after Doc, followed by all the performance related sections.

File: rust/BUILD.md [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+
+## Building evmrs (shared c library)
+
+- build `evmrs` in release mode
+    ```sh
+    cargo build --release # OR run `make` or `make tosca-rust` in Tosca directory
+    ```
+- build `evmrs` with debug symbols
+    ```sh
+    cargo build --profile profiling
```
Comment (LuisPH3) [inline rust/BUILD.md:15]: why not just cargo build? 
we may need a different explanation, is this release with symbols?

File: rust/BUILD.md [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+    ```
+
+## Lint
```
Comment (LuisPH3) [inline rust/BUILD.md:18]: It is a good practice to write a sentence between titles and enumerations. what is lint?

File: rust/BUILD.md [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+
+```sh
+cargo clippy --workspace --tests --benches --examples
+```
+
+## Doc
```
Comment (LuisPH3) [inline rust/BUILD.md:24]: please describe what are doc, where to find them once compiled.

File: rust/BUILD.md [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+
+```sh
+cargo doc --workspace --document-private-items --open
+```
+
+## Optimizations
+
+By default, `evmrs` is build with the simplest and most idiomatic version.
```
Comment (LuisPH3) [inline rust/BUILD.md:32]: version of what?  this sentence is a little loose.

File: rust/BUILD.md [hunk 8]
```diff
@@ -0,0 +1,134 @@
...
+    ```sh
+    cargo build --release # TODO remove when https://github.com/Fantom-foundation/Tosca/pull/778 is merged
+    cd ..
+    make
+    go run ./go/ct/driver run --full-mode evmrs
+    ```
+
+Also see [test.sh](./scripts/test.sh) which runs the Go tests and CT in full mode.
+
+## Benchmarking
```
Comment (LuisPH3) [inline rust/BUILD.md:63]: Can you describe how to compare benchmarks and a brief explanation of the output of this command? 
use Tosca/BUILD.md for inspiration.

**Review by LuisPH3 — COMMENTED** [review]
The overall document is a little superficial, please consider that your audience is not versed in Rust:
You may not need to describe cargo, but third party packages are obscure and need a little motivation.
- links to tools documentation are welcome.
- benchmarking has a workflow that can be described a little more in detail. 

I would say that bullet points are complete, we can work in the presentation and content of each of them.

File: rust/BUILD.md (hunk 8 above)
Comment (LorenzSchueler) [reply rust/BUILD.md:63]: I'll just link to Tosca/BUILD.md#running-benchmarks it that is fine with you.

File: rust/BUILD.md (hunk 7 above)
Comment (LorenzSchueler) [reply rust/BUILD.md:32]: ... most idiomatic implementation?

File: rust/BUILD.md (hunk 2 above)
Comment (LorenzSchueler) [reply rust/BUILD.md:115]: depends on https://github.com/Fantom-foundation/Tosca/pull/830 (renamed)

File: rust/BUILD.md (hunk 1 above)
Comment (LorenzSchueler) [reply rust/BUILD.md:51]: I guess `go test ../go/interpreter/evmrs/...` would be enough. However the go integration is not merged yet (https://github.com/Fantom-foundation/Tosca/pull/778) so those files do not yet exist.

File: rust/BUILD.md (hunk 3 above)
Comment (LorenzSchueler) [reply rust/BUILD.md:85]: will remove it in the meantime

File: rust/BUILD.md (hunk 2 above)
Comment (LorenzSchueler) [reply rust/BUILD.md]: is merged

File: rust/BUILD.md (hunk 8 above)
Comment (LuisPH3) [reply rust/BUILD.md:98]: can you compare the outputs using benchstat?

Comment (LorenzSchueler) [reply rust/BUILD.md:98]: Yes with a workaround. Have documented it now.

**Review by LuisPH3 — APPROVED** [review]

**Pull request merged**